pkg/controller/clusterconnection/clusterconnection_controller.go

```diff
@@ -145,6 +154,21 @@ func (r *ReconcileConnection) Reconcile(request reconcile.Request) (reconcile.Re
 			operatorv1.ClusterManagementTypeManaged))
 	}
 
+	// Verify that the manager is not running.
+	_, err = controllermanager.GetManager(ctx, r.Client)
```

Comment:
I suggest checking for cluster type managed instead of checking that the manager is not running.

Reply:
You are right. Right now there is no way you want guardian if it is not a managed cluster.